### ISSUE TITLE
Setting default_role breaks strict Numpydoc compatibility.

### ISSUE THREAD

**Issue by weaverba137** [astropy/sphinx-astropy#58]
[The Numpy Style Guide](https://numpydoc.readthedocs.io/en/latest/format.html#parameters) suggests the use of single backticks to refer to function parameters.

However, because sphinx-astropy (sphinx_astropy.conf.default_role) defines `default_role = 'obj'`, these are interpreted as objects rather than as (informal) references to function parameters.

For example, if I write:
```python
def rectangle(width, height=None):
    """Define a rectangle; `width` should be specified in degrees.
    """
    if height is None:
        height = width
    return (width, height)
```
I'll get a warning like: `WARNING: py:obj reference target not found: width`.

This isn't documented in [the Astropy Documentation Guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines).

See also the [first note on Roles](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#roles) in the Sphinx documentation.

**Comment by pllim** [comment]
But that section is about "Parameters" with your example does not have. Does it work if you put that reference inside "Parameters" section?

**Comment by weaverba137** [comment]
No it does not. I encountered problems with both cases.

**Comment by weaverba137** [comment]
If you like a more thorough test case would be:
```python
def rectangle(width, height=None):
    """Define a rectangle; `width` should be specified in degrees.

    Parameters
    ----------
    width : float
        Width of the rectangle in degrees.
    height : float, optional
        Height of the rectangle. If not specified, `width` will be used.

    Returns
    -------
    tuple
        The desired rectangle.
    """
    if height is None:
        height = width
    return (width, height)
```

**Comment by pllim** [comment]
And that works if you don't use sphinx-astropy? I never knew you could reference a variable.

**Comment by weaverba137** [comment]
Correct. I use Sphinx with Numpydoc, but without sphinx-astropy, all the time in other projects, and never see a warning about this.  Specifically we do not set `default_role`, rather than setting it to some other value.

**Comment by pllim** [comment]
Well, I really don't know when anyone would have time to fix this, especially given the exploration of a completely different theme at https://github.com/astropy/astropy/pull/14477 . The easiest workaround, if you have not found a better one already, is to use double backticks to avoid the ref linking. We apologize for any inconvenience caused!

**Comment by bsipocz** [comment]
I don't see what it has to do with the theme. But the problem is certainly well known but not clear anyone will have the time to fix it, we all very much used to double ticking everything that are not links. (and recently there has been some bugs in this on numpyland, too)

**Comment by pllim** [comment]
I don't think we'll be using anything from here anymore if we switch to pydata-sphinx-theme? Anyways, I am not 100% sure.

**Comment by bsipocz** [comment]
automodapi may still need some of these configs, let alone what core does has little to do with e.g. the coordinated packages.

Either case, using double backticks is the suitable workaround, or overriding the config item for the given package, but even then it may not work as expected.
